build/org.eclipse.pde.build/src/org/eclipse/pde/internal/build/site/PDEState.java

```diff
@@ -537,6 +537,10 @@ public static String querySystemPackages(IExecutionEnvironment environment, Prop
 					javaPackages = COMMA.splitAsStream(profileSystemPackages).filter(p -> p.startsWith("java.")).toList(); //$NON-NLS-1$
 				}
 				IVMInstall targetVM = JavaRuntime.getDefaultVMInstall(); // Set by the Target-Definition if specified there
+				if (targetVM == null) {
+					LOGGER.warn("No default JRE installation selected available"); //$NON-NLS-1$  
+					return null;
```

Comment:
Returning null will just move NPE to `getSystemPackages()` where this null will be immediately dereferenced. Here and below, instead of null one should return empty string.

Reply:
Sorry, after opening this in the the IDE that we are inside lambda block.

Reply:
Good point. Thanks.
I changed this and all other two locations that returned null to return an empty list, which will later result in an empty string for the list of packages.

This was forgotten in https://github.com/eclipse-pde/eclipse.pde/pull/714.